airbyte-integrations/connectors/source-mysql/src/test-integration/java/io/airbyte/integrations/source/mysql/AbstractMySqlSourceDatatypeTest.java

```diff
@@ -0,0 +1,434 @@
+        TestDataHolder.builder()
+            .sourceType("int")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .fullSourceDataType("int unsigned")
+            .addInsertValues("3428724653")
+            .addExpectedValues("3428724653")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("int")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .fullSourceDataType("int zerofill")
+            .addInsertValues("1")
+            .addExpectedValues("1")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("bigint")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .addInsertValues("null", "9223372036854775807")
+            .addExpectedValues(null, "9223372036854775807")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("float")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .addInsertValues("null", "10.5")
+            .addExpectedValues(null, "10.5")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("double")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .addInsertValues("null", "power(10, 308)", "1/power(10, 45)", "10.5")
+            .addExpectedValues(null, String.valueOf(Math.pow(10, 308)), String.valueOf(1 / Math.pow(10, 45)), "10.5")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("decimal")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .fullSourceDataType("decimal(10,3)")
+            .addInsertValues("0.188", "null")
+            .addExpectedValues("0.188", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("decimal")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .fullSourceDataType("decimal(19,2)")
+            .addInsertValues("1700000.01")
+            .addExpectedValues("1700000.01")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("date")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'2021-01-01'")
+            .addExpectedValues(null, "2021-01-01T00:00:00Z")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("datetime")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'2005-10-10 23:22:21'")
+            .addExpectedValues(null, "2005-10-10T23:22:21.000000Z")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("timestamp")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'2021-01-00'", "'2021-00-00'", "'0000-00-00'")
+            .addExpectedValues(null, null, null, null)
```

Comment:
@etsybaev is this test case intended? It seems problematic to null out all timestamps.